### PR TITLE
List show total

### PR DESCRIPTION
**Pull request by mattiaslundberg**
It's common to report the total worked hours to other systems, this is
intended to help retreiving that data easily.

I have opted to extend the list command for ease of use, it's probably
what you check anyway when reporting.

image

Last week we had some slack discussions where this should live (here + ui or api or new service). Keeping it here and ui for now, simple enough to move later.

### CODE REVIEW

**Comment by mattiaslundberg** [comment]
Test fail due to travis build/deploy on both branch and pr at the same time.

File: chalicelib/action.py [hunk 1]
```diff
@@ -465,10 +441,76 @@ def perform_action(self):
...
+                text=f"Date: *{event_date}*\nReason: *{reason}*\nHours: *{hours}*"
+            )
+            self.slack.add_divider_block()
+
+    def _show_period_data(self, list_data, period_data) -> Dict[str, int]:
+        if not period_data:
+            self.slack.add_section_block(text="No information about worked hours")
+            return
+
+        total_per_type = defaultdict(int)
```
Comment (parberge) [inline chalicelib/action.py:482]: Care to explain why we need to use `defaultdict`?

File: chalicelib/lib/period_data.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import logging
+
+import requests
+
+from chalicelib.lib.helpers import month_range, parse_date
+
+log = logging.getLogger(__name__)
+
+
+def get_period_data(date_str):
```
Comment (parberge) [inline chalicelib/lib/period_data.py:10]: Missing type hints

File: chalicelib/action.py (hunk 1 above)
Comment (mattiaslundberg) [reply chalicelib/action.py:482]: With defaultdict we can do this:
```python
total_per_type[event.get("reason")] += hours
```

Without we have to do something like this:
```python
reason = event.get("reason")
total_per_type[reason] = total_per_type.get(reason, 0) + hours
```

I prefer the first one for readability of the code

**Pull request merged**